src/utils/bods.js

```diff
@@ -1,5 +1,42 @@
 import { compareVersions } from 'compare-versions';
 export const closedRecords = new Set();
+export const changedRecords = new Set();
+
+export const getDates = (statements) => {
+  const uniqueDates = new Set();
+
+  for (const statement of statements) {
+    if (!uniqueDates.has(statement.statementDate)) {
+      uniqueDates.add(statement.statementDate);
+    }
+  }
+
+  const arr = Array.from(uniqueDates);
+  return arr.sort((a, b) => {
+    return a - b;
+  });
+};
+
+export const filteredData = (statements, version) => {
+  if (compareVersions(version, '0.4') >= 0) {
+    // get all statements with matching recordId and recordType values
+    const recordIdCount = {};
+
+    for (const statement of statements) {
+      const key = `${statement.recordId}-${statement.recordType}`;
+      recordIdCount[key] = (recordIdCount[key] || 0) + 1;
+    }
+
+    const duplicates = statements.filter((statement) => {
+      const key = `${statement.recordId}-${statement.recordType}`;
+      return recordIdCount[key] > 1;
+    });
+
+    console.log(duplicates);
+  } else {
+    // get all statements with statementID values in replacesStatements array
```

Comment:
What is the deal with this? Does this PR add rendering change over time just for BODS 0.4? If so probably best to be explicit about that in this comment?

Reply:
> Does this PR add rendering change over time just for BODS 0.4?

Yes, the [specification](https://docs.google.com/document/d/15YBQbPixnS1i1qe4RlXfkx8wchVsvuEI3ZrCS_pOg5A/edit?tab=t.0) states that only 0.4 data renders an interactive slider. I'll update the comment (and remove the `console.log`).